packages/altair-api/src/auth/user/user.service.spec.ts

```diff
@@ -2,19 +2,230 @@ import { Test, TestingModule } from '@nestjs/testing';
 import { PrismaService } from 'nestjs-prisma';
 import { StripeService } from 'src/stripe/stripe.service';
 import { UserService } from './user.service';
+import { SignupInput } from '../models/signup.input';
+import {
+  mockPlanConfig,
+  mockPrismaConflictError,
+  mockUser,
+  mockUserPlan,
+} from '../mocks/prisma-service.mock';
+import {
+  mockStripeCustomer,
+  mockSubscriptionItem,
+} from '../mocks/stripe-service.mock';
+import Stripe from 'stripe';
 
 describe('UserService', () => {
   let service: UserService;
+  let stripeService: StripeService;
+  let prismaService: PrismaService;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       providers: [UserService, PrismaService, StripeService],
     }).compile();
 
     service = module.get<UserService>(UserService);
+    stripeService = module.get<StripeService>(StripeService);
+    prismaService = module.get<PrismaService>(PrismaService);
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  describe('createUser', () => {
+    let payloadMock: SignupInput;
+
+    beforeAll(() => {
+      payloadMock = { email: 'john.doe@email.com' };
+    });
+
+    beforeEach(() => {
+      jest
+        .spyOn(stripeService, 'connectOrCreateCustomer')
+        .mockResolvedValueOnce(mockStripeCustomer());
+    });
+
+    it('should return the created user object', async () => {
+      // GIVEN
+      jest
+        .spyOn(prismaService.user, 'create')
+        .mockResolvedValueOnce(mockUser());
+
+      // WHEN
+      const user = await service.createUser(payloadMock);
+
+      // THEN
+      (expect(user) as any).toBeUser();
+    });
+
+    it('should throw and exception if the email is already taken', () => {
+      // GIVEN
+      jest
+        .spyOn(prismaService.user, 'create')
+        .mockRejectedValueOnce(mockPrismaConflictError());
+
+      // THEN
+      expect(service.createUser(payloadMock)).rejects.toThrow(
+        'Email john.doe@email.com already used.'
+      );
+    });
+
+    it('should rethrow an unknown error', () => {
+      // GIVEN
+      jest
+        .spyOn(prismaService.user, 'create')
+        .mockRejectedValueOnce(Error('Unexpected error'));
+
+      // THEN
+      expect(service.createUser(payloadMock)).rejects.toThrow(
+        'Unexpected error'
+      );
+    });
+  });
+
+  describe('mustGetUser', () => {
+    it('should return a user object', async () => {
+      // GIVEN
+      const userMock = mockUser();
+      jest.spyOn(service, 'getUser').mockResolvedValueOnce(userMock);
```

Comment:
So instead of mocking the method on the `service` being tested, let's mock the underlying `prismaService` method instead.